tests/virtualization/vagrant/boxes/tumbleweed.pm

```diff
@@ -0,0 +1,99 @@
+# Maintainer: dancermak <dcermak@suse.com>
+
+use strict;
+use warnings;
+use base "basetest";
+use testapi;
+use utils;
+use File::Basename;
+
+
+sub run() {
+    select_console('root-console');
+    zypper_call('in ansible');
+
+    select_console('user-console');
+
+    # version = Tumbleweed, Leap 15 etc
+    my $version = get_required_var('VERSION');
+    my $arch    = get_required_var('ARCH');
+    # FIXME: don't hardcode the build in production
+    # my $build = get_required_var('BUILD');
+    my $build = "20190721";
+
+    # This is a constant defined in the KIWI config file, we don't expect it to change
+    my $boxversion = '1.0';
+
+    my %boxes = (
+        # openSUSE-Tumbleweed-Vagrant.x86_64-1.0-{libvirt|virtualbox}-Snapshot20190704.vagrant.{libvirt|virtualbox}.box
+        libvirt    => "openSUSE-$version-Vagrant.$arch-$boxversion-libvirt-Snapshot$build.vagrant.libvirt.box",
+        virtualbox => "openSUSE-$version-Vagrant.$arch-$boxversion-virtualbox-Snapshot$build.vagrant.virtualbox.box"
+    );
+
+    my @providers = keys %boxes;
+
+    #
+    # get Vagrantfile template and replace the distro name & insert box filenames
+    #
+    assert_script_run("wget --quiet " . data_url("virtualization/Vagrantfile"));
+    assert_script_run("sed -i 's|DISTRO|$version|' Vagrantfile");
+
+    foreach my $provider (@providers) {
+        my $boxname = "$boxes{$provider}";
+        assert_script_run("wget --quiet " . autoinst_url("/assets/other/$boxname"));
+
+        my $upcase_provider = uc $provider;
+        assert_script_run("sed -i 's|BOXNAME_$upcase_provider|$boxname|' Vagrantfile");
+    }
+
+    # move the Vagrantfile into a empty subdirectory and invoke vagrant from
+    # there, so that we don't synchronize the huge .box files into the VM
+    assert_script_run("mkdir test_dir");
+    assert_script_run("mv Vagrantfile test_dir/");
+    assert_script_run("pushd test_dir");
+
+    #
+    # Grab the remaining test files and bring the boxes up, down and up again
+    # be sure to clean them up afterwards
+    #
+    assert_script_run("wget --quiet " . data_url('virtualization/testfile.txt'));
+    assert_script_run("wget --quiet " . data_url('virtualization/provision.sh'));
+    assert_script_run("wget --quiet " . data_url('virtualization/ansible_playbook.yml'));
+
+    foreach my $provider (@providers) {
+        my $boxname = "$version-$provider";
+
+        # Test the box *only*: bring it up and destroy it immediately afterwards
+        assert_script_run("vagrant up $boxname --provider $provider --no-provision", timeout => 1200);
+
+        # now run the actual tests via the Ansible test playbook
+        assert_script_run("vagrant provision $boxname", timeout => 1200);
+
+        # test if the box survives a reboot
+        assert_script_run("vagrant halt");
+        assert_script_run("vagrant up $boxname", timeout => 1200);
+
+        assert_script_run("vagrant destroy -f $boxname");
+    }
+
+    assert_script_run("popd");
```

Comment:
Remove the boxes at the end to save some disk space

Reply:
Done, I've also added a `vagrant box remove` to cleanup the boxes that vagrant saves.